core/controllers/questions_list.py

```diff
@@ -63,10 +67,23 @@ def get(self, comma_separated_skill_ids):
 
         (
             question_summaries, merged_question_skill_links,
-            next_start_cursor) = (
-                question_services.get_displayable_question_skill_link_details(
-                    constants.NUM_QUESTIONS_PER_PAGE, skill_ids, start_cursor)
-            )
+            next_offset
+        ) = (
+
+            question_services.get_displayable_question_skill_link_details(
+                constants.NUM_QUESTIONS_PER_PAGE, skill_ids, offset=offset)
+        )
+
+        _, _, temp_next_offset = (
+            question_services.get_displayable_question_skill_link_details(
+                1, skill_ids, offset=next_offset)
+        )
+
+        # Set next_offset to None if there are no more questions so that
+        # in the frontend we know when we reached the end of the list.
+        if next_offset == temp_next_offset:
+            next_offset = None
+
```

Comment:
What about this? And maybe above the call to `get_displayable_question_skill_link_details` explain that we fetch + 1 because we need to check if there are more models.

```suggestion
            next_offset
        ) = (
            question_services.get_displayable_question_skill_link_details(
                constants.NUM_QUESTIONS_PER_PAGE + 1, skill_ids, offset=offset)
        )
        next_offset -= 1
        
        # Set next_offset to None if there are no more questions so that
        # in the frontend we know when we reached the end of the list.
        if len(question_summaries) <= constants.NUM_QUESTIONS_PER_PAGE:
        	next_offset = None
            

```